File: C++ solutions/1512. Number of Good Pairs.cpp
// approach :1
class Solution {
public:
    int numIdenticalPairs(vector<int>& nums) {
        int cnt=0;
        for(int i=0;i<nums.size();i++){
            for(int j=i+1;j<nums.size();j++){
                if(nums[i]==nums[j]) cnt++;
            }
        }
        return cnt;
    }
};

// approach :2
class Solution {
public:
    int numIdenticalPairs(vector<int>& nums) {
        map<int,int>m;
        for(int i=0;i<nums.size();i++) m[nums[i]]++;
 
        int ans=0;
        for(auto k:m){
           int cnt=k.second-1;
           ans+=(cnt*(cnt+1))/2;
        }
        return ans;
    }
};

File: JavaScript solutions/1512. Number of Good Pairs.js
// approach :1
/**
 * @param {number[]} nums
 * @return {number}
 */
var numIdenticalPairs = function (nums) {
  let cnt = 0;
  for (let i = 0; i < nums.length; i++) {
    for (let j = i + 1; j < nums.length; j++) {
      if (nums[i] == nums[j]) cnt++;
    }
  }
  return cnt;
};

// approach :2
/**
 * @param {number[]} nums
 * @return {number}
 */
var numIdenticalPairs = function (nums) {
  const map = new Map();
  for (let i = 0; i < nums.length; i++) {
    if (!map.has(nums[i])) {
      map.set(nums[i], 1);
    } else {
      map.set(nums[i], map.get(nums[i]) + 1);
    }
  }

  let ans = 0;
  for (const count of map.values()) {
    const cnt = count - 1;
    ans += (cnt * (cnt + 1)) / 2;
  }
  return ans;
};
